.github/workflows/test-e2e.yml

```
@@ -1,143 +1,77 @@
-          fetch-depth: 0
-
-      - name: Prepare tag
-        id: prepare
-        uses: ./.github/actions/set-tag
-      - name: Log in to ghcr.io registry
-        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u $ --password-stdin
-
-      - name: Build and Push image
-        uses: ./.github/actions/build-push-image
-        with:
-          file: fast.Dockerfile
-          repository: ghcr.io/${{ env.REPOSITORY }}
-          version: ${{ steps.prepare.outputs.tag }}
-          tags: ghcr.io/${{ env.REPOSITORY }}:${{ steps.prepare.outputs.tag }}
-          platforms: linux/amd64
-          push_to_docker: false
-
-      - name: Do preflight-check on test image
-        uses: ./.github/actions/certify-openshift-images
-        with:
-          registry: ghcr.io
-          registry_password: ${{ secrets.GITHUB_TOKEN }}
-          repository: ${{ env.REPOSITORY }}
-          version: ${{ steps.prepare.outputs.tag }}
-          rhcc_token: ${{ secrets.RH_CERTIFICATION_PYXIS_API_TOKEN }}
-          rhcc_project: ${{ secrets.RH_CERTIFICATION_OSPID }}
-          submit: false
-
-  prepare-e2e-bundle:
-    name: Prepare E2E Bundle configuration and image
-    runs-on: ubuntu-latest
+      image_url: ${{ steps.set_image_url.outputs.image_url }}
     env:
-      GHCR_REPO: ghcr.io/mongodb/mongodb-atlas-kubernetes-operator-prerelease
-      GHCR_BUNDLES_REPO: ghcr.io/mongodb/mongodb-atlas-kubernetes-bundles-prerelease
+      REPO: ghcr.io/${{ github.repository_owner }}/mongodb-atlas-kubernetes-operator-prerelease
     steps:
-      - name: Check out code
-        uses: actions/checkout@v4
+      - uses: actions/checkout@v4
         with:
-          ref: ${{github.event.pull_request.head.sha}}
-          submodules: true
-          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
 
-      - name: Prepare tag
-        id: prepare
+      - id: set_tag
         uses: ./.github/actions/set-tag
 
-      - name: Generate configuration for the tests
-        uses: ./.github/actions/gen-install-scripts
-        with:
-          IMAGE_URL: ${{ env.GHCR_REPO }}:${{ steps.prepare.outputs.tag }}
-          VERSION: ${{ steps.prepare.outputs.tag }}
-          ENV: dev
-
-      - name: Change path for the test
+      - id: set_image_url
         run: |
-          awk '{gsub(/cloud.mongodb.com/, "cloud-qa.mongodb.com", $0); print}' bundle/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml > tmp && mv tmp bundle/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml
+          echo "image_url=${REPO}:${{ steps.set_tag.outputs.tag }}" >> "$GITHUB_OUTPUT"
 
-      - name: Cache repo files
-        uses: actions/cache@v4
-        with:
-          path: |
-            ./*
-          key: ${{ github.sha }}
+      - run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" \
+          | docker login ghcr.io \
+              -u ${{ github.actor }} \
+              --password-stdin
```

Let's use the action instead of "hand write" it
```suggestion
      - name: Log in to the GitHub Container Registry
        uses: docker/login-action@v3
        with:
          registry: ghcr.io
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}
```

---

This makes sense. I will work on it now!